discretisedfield/mesh.py

```diff
@@ -950,6 +950,123 @@ def plane(self, *args, n=None, **kwargs):
 
         return plane_mesh
 
+    def sel(self, *args, **kwargs):
+        if len(args) > 1 or len(kwargs) > 1:
+            raise ValueError("Select method only accepts one dimension at a time.")
+
+        if args and not kwargs:
+            dim = args[0]
+            range_ = None
+        elif not args and kwargs:
+            dim, range_ = list(kwargs.items())[0]
+        else:
+            raise ValueError(
+                "Either one positional argument or a keyword argument can be passed."
+            )
+
+        dim_index = self.region._dim2index(dim)
+
+        if range_ is not None:
+            if isinstance(range_, numbers.Real):
+                # TODO: Some book-keeping in future.
+                selected_value = range_
+            elif isinstance(range_, (tuple, list, np.ndarray)):
+                if len(range_) != 2 or not all(
+                    isinstance(point, numbers.Real) for point in range_
+                ):
+                    raise ValueError(
```

Comment:
Should `range_ = (0, 'z')` give a `TypeError`?

Reply:
I can split this if statement in two and raise `ValueError` if `len(range_)` is not two and `TypeError` if the values are not `numbers.Real`?